infrastructure/src/main/java/org/corfudb/infrastructure/CompactorService.java

```diff
@@ -135,51 +189,51 @@ private void runOrchestrator() {
-                        } else {
-                            getCompactorLeaderServices().validateLiveness();
-                        }
-                    } else if (compactionTriggerPolicy.shouldTrigger(
-                            getCorfuRuntime().getParameters().getCheckpointTriggerFreqMillis(), getCorfuStore())) {
-                        trimLog.invokePrefixTrim();
-                        compactionTriggerPolicy.markCompactionCycleStart();
-                        getCompactorLeaderServices().initCompactionCycle();
+            if (isLeader) {
+                if (managerStatus != null && managerStatus.getStatus() == StatusType.STARTED) {
+                    if (getDistributedCheckpointerHelper().isCompactionDisabled()) {
+                        log.info("Compaction has been disabled. Force finish compaction cycle as it already started");
+                        compactorLeaderServices.finishCompactionCycle();
+                    } else {
+                        compactorLeaderServices.validateLiveness();
                     }
+                } else if (compactionTriggerPolicy.shouldTrigger(
+                        this.corfuRuntimeParameters.getCheckpointTriggerFreqMillis(), getCorfuStore())) {
+                    trimLog.invokePrefixTrim(getCorfuRuntime(), getCorfuStore());
+                    compactionTriggerPolicy.markCompactionCycleStart();
+                    compactorLeaderServices.initCompactionCycle();
                 }
-                if (managerStatus != null) {
-                    if (managerStatus.getStatus() == StatusType.FAILED || managerStatus.getStatus() == StatusType.COMPLETED) {
-                        checkpointerJvmManager.shutdown();
-                    } else if (managerStatus.getStatus() == StatusType.STARTED && !checkpointerJvmManager.isRunning()
-                            && !checkpointerJvmManager.isInvoked()) {
-                        checkpointerJvmManager.invokeCheckpointing();
-                    }
+            }
+            if (managerStatus != null) {
+                if (managerStatus.getStatus() == StatusType.FAILED || managerStatus.getStatus() == StatusType.COMPLETED) {
+                    checkpointerJvmManager.shutdown();
+                } else if (managerStatus.getStatus() == StatusType.STARTED && !checkpointerJvmManager.isRunning()
+                        && !checkpointerJvmManager.isInvoked()) {
+                    checkpointerJvmManager.invokeCheckpointing();
                 }
-            } catch (Exception ex) {
-                log.warn("Exception in runOrcestrator(): ", ex);
             }
+        } catch (Exception ex) {
+          log.error("Exception in runOrchestrator(): ", ex);
         } catch (Throwable t) {
-            log.error("Encountered unexpected exception in runOrchestrator: ", t);
-            throw t;
+            log.error("Encountered unexpected exception in runOrchestrator(): ", t);
```

Comment:
Can we change the error message to distinguish this case, when just a throwable is being handled. Ideally this case of handling a throwable should never occur right?